ocs/cluster.py

```diff
@@ -0,0 +1,248 @@
+CEPHCLUSTER = ocp.OCP(
+    kind='CephCluster', namespace=ENV_DATA['cluster_namespace']
+)
+
+logger = logging.getLogger(__name__)
+
+
+class CephCluster(object):
+    """
+    Handles all cluster related operations from ceph perspective
+
+    This class has depiction of ceph cluster. Contains references to
+    pod objects which represents ceph cluster entities.
+
+    Attributes:
+        _ocs_pods (list) : A list of  ceph cluster related pods
+        _cluster_name (str): Name of ceph cluster
+        _namespace (str): openshift Namespace where this cluster lives
+        _api_client (APIClient): api-client used for interacting with openshift
+        _url_prefix (str): url prefix for REST api interaction
+    """
+
+    def __init__(self, **config):
+        # cluster_name is name of cluster in rook of type CephCluster
+        self._cluster_name = config.get('cluster_name', 'rook-ceph')
+        """
+        TODO: Instantiate CephCluster object somewhere higher after deployment
+        Read namespace and cluster name from config.
+        """
+        self._namespace = config.get(
+            'namespace', ENV_DATA['cluster_namespace']
+        )
+        self._ocs_pods = pod.get_all_pods(self._namespace)
+
+        #self.cluster_resource = os.path.join(
+        #    default.TEMPLATE_DIR, "cluster.yaml")
+        # Once we figure out where to initialize this CephCluster object we
+        # will pass **config
+        #self.cluster_config = generate_yaml_from_jinja2_template_with_data(
+        #    self.cluster_resource, **config
+        #)
+        self.cluster_config = CEPHCLUSTER.get()['items'][0]
+        self.cephfs_config = default.CEPHFILESYSTEM_DICT.copy()
+        #Below is just for testing purpose
+        #self.cephfs_config['metadata']['name'] = 'cephfs-test-0913235459'
+
+        # We are not invoking ocs.create() here
+        # assuming cluster creation is done somewhere after deployment
+        # So just load ocs with existing cluster details
+        self.cluster = ocs.OCS(**self.cluster_config)
+        self.cephfs = ocs.OCS(**self.cephfs_config)
+
+        self.cluster.reload()
+        self.cephfs.reload()
+
+        self.mon_selector = default.MON_APP_LABEL
+        self.mds_selector = default.MDS_APP_LABEL
+        self.tool_selector = default.TOOL_APP_LABEL
+
+        self.mons = []
+        self.mdss = []
+        self.toolbox = []
+
+        self.scan_cluster()
+        self.mon_count = len(self.mons)
+        self.mds_count = len(self.mdss)
+
+    @property
+    def cluster_name(self):
+        return self._cluster_name
+
+    @property
+    def namespace(self):
+        return self._namespace
+
+    @property
+    def pods(self):
+        return self._ocs_pods
+
+    def scan_cluster(self):
+        """
+        Get accurate info on current state of pods
+        """
+        self._ocs_pods = pod.get_all_pods(self._namespace)
+        self.mons = self._filter_pods(self.mon_selector, self.pods)
```

Comment:
I think that having a functions for get_mds_pods(), get_mon_pods() and get_mgr_pods() will be good to be placed under pod.py as helpers and you will be able to use those.
WDYT?

Reply:
@RazTamir 
Any thoughts on keeping it in class or making methods?

Reply:
ah ok, I got it. I will do that

Reply:
Done